src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -177,6 +184,10 @@ public function normalize($data, string $format = null, array $context = [])
      */
     public function denormalize($data, string $type, string $format = null, array $context = [])
     {
+        if (isset(self::SCALAR_TYPES[$type]) && \call_user_func('is_'.$type, $data)) {
```

Comment:
What about throwing a nicer exception in case `$type` is once of the reference scalar types but `\call_user_func('is_'.$type, $data)` is `false` ? That would make more sense than getting an exception about adding a normalizer for an object.